orangecontrib/spectroscopy/io/neaspec.py

```diff
@@ -11,195 +11,285 @@
-
-            final_metas = []
-            final_data = []
-
-            for row, col in uniquerc:
-                cur = di[(row, col)]
-                M, OA, OP = cur["M"], cur["OA"], cur["OP"]
-
-                OAn = np.zeros(OA.shape) * np.nan
-                OPn = np.zeros(OA.shape) * np.nan
-                for run in range(len(M)):
-                    f = interp1d(M[run], OA[:, run])
-                    OAn[:, run] = f(X)
-                    f = interp1d(M[run], OP[:, run])
-                    OPn[:, run] = f(X)
-
-                OAmean = np.mean(OAn, axis=1)
-                OPmean = np.mean(OPn, axis=1)
-                final_data.append(OAmean)
-                final_data.append(OPmean)
-                final_metas += [[row, col, "O%dA" % i] for i in range(numharmonics)]
-                final_metas += [[row, col, "O%dP" % i] for i in range(numharmonics)]
-
-            final_data = np.vstack(final_data)
-
-            metas = [Orange.data.ContinuousVariable.make("row"),
-                     Orange.data.ContinuousVariable.make("column"),
-                     Orange.data.StringVariable.make("channel")]
-
-            domain = Orange.data.Domain([], None, metas=metas)
-            meta_data = Table.from_numpy(domain, X=np.zeros((len(final_data), 0)),
-                                         metas=np.asarray(final_metas, dtype=object))
-            return X, final_data, meta_data
-
-    def read_v2(self):
-
-        # Find line in which data begins
-        count = 0
-        with open(self.filename, "rt", encoding="utf8") as f:
-            while f:
-                line = f.readline()
-                count = count + 1
-                if line[0] != '#':
-                    break
-
-            file = np.loadtxt(f)  # Slower part
-
-        # Find the Wavenumber column
-        line = line.strip().split('\t')
-
-        for i, e in enumerate(line):
-            if e == 'Wavenumber':
-                index = i
-                break
-
-        # Channel need to have exactly 3 letters
-        Channel = line[index + 1:]
-        Channel = np.array(Channel)
-        # Extract other data #
-        Max_row = int(file[:, 0].max() + 1)
-        Max_col = int(file[:, 1].max() + 1)
-        Max_omega = int(file[:, 2].max() + 1)
-        N_rows = Max_row * Max_col * Channel.size
+        # Run is only there for ifg files
+        if "Run" in list(data.keys()):
+            Max_run = int(np.max(data["Run"]) + 1)
+        else:
+            Max_run = int(1)
+
+        if self.filename.endswith(".nea"):
+            Max_omega = int(measparams["PixelArea"][2])
+        else:
+            # Neaspec have several the naming
+            # for the indepedent variable index
+            if "Depth" in list(data.keys()):
+                Max_omega = int(np.max(data["Depth"]) + 1)
+            elif "Index" in list(data.keys()):
+                Max_omega = int(np.max(data["Index"]) + 1)
```

Comment:
Why int(n+1) ? I saw these were present before, but why?

If for rounding, this is a particularly bad way to do it. The reason deserves a comment in code.

Reply:
Also, the code is full of needless conversion to list when searching in a dictionary.